SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -196,14 +212,126 @@ void USpatialNetDriver::OnConnected()
 	Dispatcher->Init(this);
 	Sender->Init(this);
 	Receiver->Init(this, TimerManager);
-	GlobalStateManager->Init(this);
+	GlobalStateManager->Init(this, TimerManager);
+	SnapshotManager->Init(this);
+
+	// If we're the client, we can now ask the server to spawn our controller.
+	if (ServerConnection)
+	{
+		// If we know the GSM is already accepting players, simply spawn.
+		if (GlobalStateManager->bAcceptingPlayers)
+		{
+			PlayerSpawner->SendPlayerSpawnRequest();
+		}
+		else
+		{
+			// Bind a delegate to spawn this player if AcceptingPlayers changes to true.
+			auto PlayerSpawnerRef = TWeakObjectPtr<USpatialPlayerSpawner>(PlayerSpawner);
+			GlobalStateManager->AcceptingPlayersChanged.BindLambda([PlayerSpawnerRef](bool bAcceptingPlayers) {
```

Comment:
Change this to a flag on the `NetDriver` to indicate whether it's waiting for the `AcceptingPlayers` to turn true.
GSM will then call a function on the `NetDriver` when `AcceptingPlayers` turns true, and that will call `SendPlayerSpawnRequest`.